lib/utils.js

```diff
@@ -187,6 +203,38 @@ var normalizeHintField = function normalizeHintField(hint) {
   return finalHint;
 };
 
```

Comment:
I suspect the changes in this file will move given previous comments, but I'll still review them anyway: